### PR TITLE
Fixes for RWBuffer/groupshared to have right Format and StructureByte…

### PR DESCRIPTION
**Pull request by paigeale**
…Size

Having the format not set correctly was causing issues in striding and causing the float atomics test to fail.

### CODE REVIEW

File: tools/clang/test/HLSL/ShaderOpArith.xml [hunk 1]
```diff
@@ -2063,11 +2063,11 @@
       <Descriptor Name="U1" Kind="UAV" ResName="U1"
                   NumElements="64" StructureByteStride="4" />
       <Descriptor Name="U2" Kind="UAV" ResName="U2"
-                  NumElements="64" StructureByteStride="4" />
+                  NumElements="64" Format="R32_FLOAT" StructureByteStride="0" />
       <Descriptor Name="U3" Kind="UAV" ResName="U3" Dimension="TEXTURE1D"
                   NumElements="64" StructureByteStride="4" />
```
Comment (pow2clk) [inline tools/clang/test/HLSL/ShaderOpArith.xml:2068]: In fact, I think U3 needs to be altered too. It should also have a Format="R32_FLOAT" param with zero stride. (or you can just leave that out and it defaults to zero)

**Review by pow2clk — COMMENTED** [review]
Good catch! I thought we'd gotten all of these, but clearly the float variant was overlooked. 

What you have is fine, but there is an omission I think we should correct here too.

File: tools/clang/test/HLSL/ShaderOpArith.xml (hunk 1 above)
Comment (paigeale) [reply tools/clang/test/HLSL/ShaderOpArith.xml:2068]: Hey Greg. I didn't have any issue with the U3 RWTexture1D case the test was passing. Would you like me to modify this patch to include changes to U3 as well?

Comment (paigeale) [reply tools/clang/test/HLSL/ShaderOpArith.xml:2068]: Sorry scratch my last comment, we were treating that case "U3" incorrectly. I made the changes you suggested to U3 and I am seeing it pass as well

**Review by pow2clk — APPROVED** [review]
Looks good! thanks!

**Pull request merged**